packages/cli-node/src/pacman/yarn/yarn.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { assertError, ForwardedError } from '@backstage/errors';
+import { PackageInfo, PackageManager } from '../pacman';
+import { Lockfile } from '../lockfile';
+import { fetchPackageInfo } from './packageInfo';
+import { YarnVersion } from './types';
+import { YarnLockfile } from './Lockfile';
+import { paths } from '../../paths';
+import { getHasYarnPlugin } from './plugin';
+import { runYarnInstall } from './install';
+import fs from 'fs-extra';
+import { GitUtils } from '../../git';
+import { execFile } from '../../run';
+
+export class Yarn implements PackageManager {
+  private constructor(private readonly yarnVersion: YarnVersion) {}
+
+  static async create(dir?: string): Promise<Yarn> {
+    const yarnVersion = await detectYarnVersion(dir);
+    return new Yarn(yarnVersion);
+  }
+
+  name() {
+    return 'yarn';
+  }
+
+  version() {
+    return this.yarnVersion.version;
+  }
+
+  lockfilePath(): string {
+    return 'yarn.lock';
+  }
+
+  async install(): Promise<void> {
+    await runYarnInstall();
+  }
+  async runScript(_scriptName: string): Promise<void> {
+    throw new Error('Method not implemented.');
+  }
+
+  async fetchPackageInfo(name: string): Promise<PackageInfo> {
+    return fetchPackageInfo(name, this.yarnVersion);
+  }
+
+  async loadLockfile(gitRef?: string): Promise<Lockfile> {
+    const lockfilePath = paths.resolveTargetRoot('yarn.lock');
+
+    let lockfileContents: string;
+    if (gitRef) {
+      lockfileContents = await GitUtils.readFileAtRef('yarn.lock', gitRef);
+    } else {
+      lockfileContents = await fs.readFile(lockfilePath, 'utf8');
+    }
```

Comment:
definitely on the fence about exposing this git ref API versus just exposing the `parse` method on the lockfile